Components/Graph/TestSuite/GraphTests.cpp

```diff
@@ -0,0 +1,659 @@
+#include <catch2/catch.hpp>
+#include <memory>
+
+// QuICC
+#include "Graph/Shims/MlirShims.hpp"
+#include "Graph/OpsMap.hpp"
+#include "Graph/Jit.hpp"
+#include "Graph/Tags.hpp"
+#include "Memory/Cpu/NewDelete.hpp"
+#include "Memory/Cuda/Malloc.hpp"
+#include "Memory/Memory.hpp"
+
+#include "Types/Math.hpp"
+#include "Profiler/Interface.hpp"
+
+TEST_CASE("One Dimensional Loop Fourier", "[OneDimLoopFourier]")
+{
+  // Test Graph
+  std::string modStr = R"mlir(
+    func.func @entry(%tumod: tensor<?x?x?xcomplex<f64>>) -> (tensor<?x?x?xcomplex<f64>>) {
+      %tuval = quiccir.fr.prj %tumod : tensor<?x?x?xcomplex<f64>> -> tensor<?x?x?xf64> attributes{kind = "P"}
+      %ret = quiccir.fr.int %tuval : tensor<?x?x?xf64> -> tensor<?x?x?xcomplex<f64>> attributes{kind = "P"}
+      return %ret : tensor<?x?x?xcomplex<f64>>
+    }
+  )mlir";
+
+  // Grid dimensions
+  constexpr std::uint32_t rank = 3u;
+  // v012
+  std::array<std::uint32_t, rank> physDims{5, 3, 11};
+  std::array<std::uint32_t, rank> modsDims{5, 3, 6};
+
+  // View Types
+  std::array<std::array<std::string, 2>, 3> layOpt;
+  layOpt[0] = {"DCCSC3D", "DCCSC3D"};
+  // layOpt[1] = {"DCCSC3D", "S1CLCSC3D"};
```

Comment:
Needed?

Reply:
it is more for documentation, but I can remove it if you want